pkg/recorder/recorder.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recorder
+
+import (
+	"github.com/golang/glog"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/client/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+type RecorderGetFuncType func(string, *runtime.Scheme) record.EventRecorder
```

Comment:
I don't think the interface should take the Scheme.  This should be set by the manager.Manager so users don't need to do the wiring themselves.

Reply:
If this is an interface instead of a function, we can provide the interface.  Maybe `RecoderProvider`?